owncloudApp/src/main/java/com/owncloud/android/presentation/ui/accounts/AccountManagementActivity.kt

```diff
@@ -0,0 +1,298 @@
+import com.owncloud.android.ui.dialog.RemoveAccountDialogViewModel
+import com.owncloud.android.utils.PreferenceUtils
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+import com.owncloud.android.presentation.ui.authentication.EXTRA_ACCOUNT as EXTRA_ACCOUNT_LOGIN_ACTIVITY
+
+const val KEY_CURRENT_ACCOUNT_CHANGED = "CURRENT_ACCOUNT_CHANGED"
+const val KEY_ACCOUNT_LIST_CHANGED = "ACCOUNT_LIST_CHANGED"
+
+class AccountManagementActivity : FileActivity(), AccountManagementAdapter.AccountAdapterListener, AccountManagerCallback<Boolean> {
+
+    private val accountListAdapter: AccountManagementAdapter = AccountManagementAdapter(this)
+    private lateinit var originalAccounts: Set<String>
+    private lateinit var originalCurrentAccount: String
+    private lateinit var accountBeingRemoved: String
+    private lateinit var tintedCheck: Drawable
+
+    private val removeAccountDialogViewModel: RemoveAccountDialogViewModel by viewModel()
+    private val accountsManagementViewModel: AccountsManagementViewModel by viewModel()
+
+    @Override
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.accounts_layout)
+        tintedCheck = ContextCompat.getDrawable(this, R.drawable.ic_current_white)!!
+        tintedCheck = DrawableCompat.wrap(tintedCheck)
+        val tint = ContextCompat.getColor(this, R.color.actionbar_start_color)
+        DrawableCompat.setTint(tintedCheck, tint)
+
+        val recyclerView: RecyclerView = findViewById(R.id.account_list_recycler_view)
+        recyclerView.run {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(applicationContext)
+            adapter = accountListAdapter
+            layoutManager = LinearLayoutManager(this@AccountManagementActivity)
+        }
+
+        setupStandardToolbar(
+            getString(R.string.prefs_manage_accounts),
+            displayHomeAsUpEnabled = true,
+            homeButtonEnabled = true,
+            displayShowTitleEnabled = true
+        )
+
+        val accountList = accountsManagementViewModel.getLoggedAccounts()
+        originalAccounts = toAccountNameSet(accountList)
+        originalCurrentAccount = accountsManagementViewModel.getCurrentAccount()?.name.toString()
+
+
+        accountListAdapter.submitAccountList(accountList = getAccountListItems())
+
+        account = accountsManagementViewModel.getCurrentAccount()
+        onAccountSet(false)
+
+    }
+
+    /**
+     * converts an array of accounts into a set of account names.
+     *
+     * @param accountList the account array
+     * @return set of account names
+     */
+    private fun toAccountNameSet(accountList: Array<Account>): Set<String> {
+        val actualAccounts: MutableSet<String> = HashSet(accountList.size)
+        for (account in accountList) {
+            actualAccounts.add(account.name)
+        }
```

Comment:
This is totally fine, but can be also more kotlinized by using the `forEach` expression... do you dare to try it? 😛

Reply:
Better this way? 
 ```
 accountList.forEach{ account ->
            actualAccounts.add(account.name)
        }
```